executors/rust/Cargo.toml

```diff
@@ -12,15 +12,29 @@ env_logger = "0.9.1"
 
 substring = "1.0"
 json = "0.12.4"
-serde = { version = "1.0.143", features = ["derive"] }
+serde = {version = "1.0.143", features = ["derive", "alloc"], optional=true}
 serde_json = "1.0.48"
 rustc_version_runtime = "0.1.*"
     
-fixed_decimal = {version="0.5.0"}
-writeable =  {version="0.5.0"}
-
-icu = { version="1.0.0", features = ["serde"] }
-icu_testdata = { version="1.0.0", features = ["metadata"] }
-icu_provider = { version="1.0.0"}
-
+fixed_decimal = {version="0.5.3"}
+writeable = {version="0.5.2"}
+icu_compactdecimal = {version="0.2.0"}
+icu_displaynames = {version="0.10.0"}
+    
+icu = {version="1.2.0", features = ["serde"] }
+icu_testdata = {version="1.2.0"}
```

Comment:
Issue: You should only need to depend on `icu@1.2`, `icu_testdata@1.2`, and maybe `fixed_decimal`, `writeable`, and `icu_locid_adapters` if you need to reference their types.